ruby/app/controllers/articles.rb

```diff
@@ -0,0 +1,61 @@
+class ArticleController
+  def create_article(article)
+    article_exists = Article.where(title: 'title').any? { |_a| true }
+
+    return { ok: false, msg: 'Article with given title already exists' } if article_exists
+
+    new_article = Article.new
+    new_article.set_fields({ title: article['title'], content: article['content'], created_at: Time.now },
+                           %i[title content created_at])
+    new_article.save_changes
+
+    { ok: false, obj: article }
+  rescue StandardError
+    { ok: false }
+  end
+
+  def update_article(id, new_data)
+    article = Article.where(id: id).first
+
+    return { ok: false, msg: 'Article could not be found' } unless article.nil?
+
+    article.set_fields({ title: new_data['title'], content: new_data['content'] }, %i[title content])
+    article.save_changes
+
+    { ok: true }
+  rescue StandardError
+    { ok: false }
+  end
+
+  def get_article(id)
+    res = Article.where(id: id)
+
+    if res.empty?
+      data = []
+
+      res.each do |r|
```

Comment:
The idea here is to change the data type from `hash` to an `array`.  Using `Hash#map` would return a hash, if I'm not mistaken. 

I could have returned hashes instead of arrays in one liners, but this makes the app look a bit more "involving" imo.

Reply:
ok